pandas/tests/arrays/sparse/test_libsparse.py

```diff
@@ -16,183 +16,134 @@
-
-        """
-        x: ----
-        y:     ----
-        r: --------
-        """
-        xloc = [0]
-        xlen = [5]
-        yloc = [5]
-        ylen = [4]
-        eloc = [0]
-        elen = [9]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
-        """
-        x: -----     -----
-        y:   -----          --
-        """
-        xloc = [0, 10]
-        xlen = [5, 5]
-        yloc = [2, 17]
-        ylen = [5, 2]
-        eloc = [0, 10, 17]
-        elen = [7, 5, 2]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
-        """
-        x: ------
-        y:    -------
-        r: ----------
-        """
-        xloc = [1]
-        xlen = [5]
-        yloc = [3]
-        ylen = [5]
-        eloc = [1]
-        elen = [7]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
-        """
-        x: ------  -----
-        y:    -------
-        r: -------------
-        """
-        xloc = [2, 10]
-        xlen = [4, 4]
-        yloc = [4]
-        ylen = [8]
-        eloc = [2]
-        elen = [12]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
-        """
-        x: ---  -----
-        y: -------
-        r: -------------
-        """
-        xloc = [0, 5]
-        xlen = [3, 5]
-        yloc = [0]
-        ylen = [7]
-        eloc = [0]
-        elen = [10]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
-        """
-        x: ------  -----
-        y:    -------  ---
-        r: -------------
-        """
-        xloc = [2, 10]
-        xlen = [4, 4]
-        yloc = [4, 13]
-        ylen = [8, 4]
-        eloc = [2]
-        elen = [15]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
-        """
-        x: ----------------------
-        y:   ----  ----   ---
-        r: ----------------------
-        """
-        xloc = [2]
-        xlen = [15]
-        yloc = [4, 9, 14]
-        ylen = [3, 2, 2]
-        eloc = [2]
-        elen = [15]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
-        """
-        x: ----       ---
-        y:       ---       ---
-        """
-        xloc = [0, 10]
-        xlen = [3, 3]
-        yloc = [5, 15]
-        ylen = [2, 2]
-        eloc = [0, 5, 10, 15]
-        elen = [3, 2, 3, 2]
-        _check_case(xloc, xlen, yloc, ylen, eloc, elen)
+    @pytest.mark.parametrize(
+        "xloc, xlen, yloc, ylen, eloc, elen",
+        [
+            [[0], [5], [5], [4], [0], [9]],
+            [[0, 10], [5, 5], [2, 17], [5, 2], [0, 10, 17], [7, 5, 2]],
+            [[1], [5], [3], [5], [1], [7]],
+            [[2, 10], [4, 4], [4], [8], [2], [12]],
+            [[0, 5], [3, 5], [0], [7], [0], [10]],
+            [[2, 10], [4, 4], [4, 13], [8, 4], [2], [15]],
+            [[2], [15], [4, 9, 14], [3, 2, 2], [2], [15]],
+            [[0, 10], [3, 3], [5, 15], [2, 2], [0, 5, 10, 15], [3, 2, 3, 2]],
+        ],
+    )
+    def test_index_make_union(self, xloc, xlen, yloc, ylen, eloc, elen):
+        # Case 1
```

Comment:
nitpick might make sense for this comment to be split into pieces between the cases L94-L101?

Reply:
OK for follow-up, this is a big improvement as is